python/triton/compiler/compiler.py

```diff
@@ -373,7 +373,8 @@ def compile(fn, **kwargs):
     _device_backend = get_backend(device_type)
 
     if device_type in ["cuda", "hip"]:
-        arch = get_architecture_descriptor(kwargs.get("cc", None))
+        # hip with kwargs.get("cc", None) causes multiprocessing issues in torch.compile
+        arch = get_architecture_descriptor(None if device_type == "hip" else kwargs.get("cc", None))
```

Comment:
I see multiple ways to fix issue with `test_dot_mfma_vector_load`:

1) change this check so it will ignore integer cc values, but take rocm compatible architecture description
2) always ignore cc option, but introduce new compile argument, specific for amd 
3) hack vector load test , so we can keep this core as-is :) 

@jataylo 
What do you think about this kind of "fast" fix? (this is first option from list below)

```suggestion
        # hip with kwargs.get("cc", None) causes multiprocessing issues in torch.compile
        if device_type == "hip":
            cc = kwargs.get("cc", None)
            arch = get_architecture_descriptor(None if type(cc) is int else cc)
        else:
            arch = get_architecture_descriptor(kwargs.get("cc", None))
```